lib/connect-backbone-to-react.js

```diff
@@ -0,0 +1,147 @@
+const hoistStatics = require('hoist-non-react-statics');
+const { Component, PropTypes, createElement } = require('react');
+const debounceFn = require('lodash.debounce');
+const isArray = require('lodash.isarray');
+
+function getDisplayName(name) {
+  return `connectBackboneToReact(${name})`;
+}
+
+function defaultMapModelsToProps(models) {
+  return Object.keys(models).reduce((acc, modelKey) => {
+    const model = models[modelKey];
+    acc[modelKey] = model.toJSON();
+    return acc;
+  }, {});
+}
+
+module.exports = function connectBackboneToReact(
+  mapModelsToProps,
+  options = {}
+) {
+  if (typeof mapModelsToProps !== 'function') {
+    mapModelsToProps = defaultMapModelsToProps;
+  }
+
+  const {
+    debounce = false,
+    events = {},
+    modelTypes = {},
+  } = options;
+
+  function getEventNames(modelName) {
+    let eventNames = events[modelName];
+
+    // Allow turning off event handlers by setting events to false.
+    if (eventNames === false) {
+      return [];
+    }
+
+    if (!isArray(eventNames)) {
+      eventNames = [];
+    }
+
+    if (eventNames.length === 0) {
+      eventNames.push('all');
+    }
+
+    return eventNames;
+  }
+
+  function validateModelTypes(modelsMap) {
+    return Object.keys(modelTypes).forEach(modelKey => {
+      const ModelConstructor = modelTypes[modelKey];
+      const modelInstance = modelsMap[modelKey];
+
+      const isInstanceOfModel = modelInstance instanceof ModelConstructor;
+      if (!isInstanceOfModel) {
+        throw new Error(`"${modelKey}" model found on modelsMap does not match type required.`);
```

Comment:
Ideally I could show the constructor name but I tried to figure out a way to sniff for that value and couldn't figure it out.

![image](https://cloud.githubusercontent.com/assets/322576/23283624/f3ac8760-f9f4-11e6-9fd3-6872a8abd172.png)